### PR TITLE
kubeadm: fix static check failures

### LINKED ISSUE #1719: fix static check failures for cmd/kubeadm
this PR will add static checking of the kubernetes/kubernetes repository:
https://github.com/kubernetes/kubernetes/pull/81189

mini-guide:
- wait for https://github.com/kubernetes/kubernetes/pull/81189 to merge.
- clone kubernetes/kubernetes or fetch the latest master.
- modify the file `./hack/.staticcheck_failures` to exclude all `cmd/kubeadm/*` entries.
- run `./hack/verify-staticcheck.sh`.
- look at the problems the check returns a send a PR to kubernetes/kubernetes to fix them.

/kind bug
/priority backlog
/help
/milestone v1.16

### PR DESCRIPTION
**Pull request by SataQiu**
**What type of PR is this?**
/kind cleanup
/kind bug

**What this PR does / why we need it**:
kubeadm: fix static check failures

**Which issue(s) this PR fixes**:

Fixes kubernetes/kubeadm#1719

Ref: #81657

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by neolit123** [comment]
/lgtm
/approve
/priority backlog
thanks @SataQiu 

/assign @BenTheElder 
for /hack approval

File: cmd/kubeadm/app/util/system/types_unix.go [hunk 1]
```diff
@@ -23,9 +23,6 @@ import (
 	"strings"
 )
 
-// dockerEndpoint is the os specific endpoint for docker communication
-const dockerEndpoint = "unix:///var/run/docker.sock"
```
Comment (neolit123) [inline cmd/kubeadm/app/util/system/types_unix.go]: i'm going to assume unused package level constants with these.

File: cmd/kubeadm/app/preflight/checks.go [hunk 2]
```diff
@@ -214,7 +214,7 @@ func (poc PortOpenCheck) Check() (warnings, errorList []error) {
 		}
 	}
 
-	return nil, errorList
+	return warnings, errorList
```
Comment (BenTheElder) [inline cmd/kubeadm/app/preflight/checks.go]: good find :-)

File: cmd/kubeadm/app/phases/certs/renewal/manager.go [hunk 3]
```diff
@@ -257,6 +257,9 @@ func (rm *Manager) CreateRenewCSR(name, outdir string) error {
 
 	// generates the CSR request and save it
 	csr, key, err := pkiutil.NewCSRAndKey(cfg)
+	if err != nil {
```
Comment (BenTheElder) [inline cmd/kubeadm/app/phases/certs/renewal/manager.go]: also a good find

**Review by BenTheElder — APPROVED** [review]
/approve
for hack
big +1

**Comment by nikhita** [comment]
PR needs rebase :(

**Comment by neolit123** [comment]
@SataQiu could you please rebase, something must have merged before this (possibly the kustomize UX pr)?

**Comment by SataQiu** [comment]
Than you for reminding me! @nikhita @neolit123 
Rebased.

**Comment by SataQiu** [comment]
ping @dixudx  Need LGTM again!

**Review by dixudx — APPROVED** [review]
/lgtm

**Pull request merged**